obspy/core/event/base.py

```diff
@@ -611,43 +667,73 @@ def get_referred_object(self):
         Will return None if no object could be found.
         """
```

Comment:
Ok, good point here but I am not sure how to do this correctly. In order to make this thread-safe(ish) I think the lock would need to be on read_events, and Catalog.copy as well wouldn't it? 

For example, if the following 3 things don't happen for each event before any occur for another event object with the same resource_ids, the "event scoped" behavior would be ruined:
1. Each non-resource_id object is created
2. The resource_id of each of the newly created objects calls set_referred_object 
3.  The _object_id attr of each any resource_ids not named "resource_id" is cleared
4. bind_resource_ids gets called 

If I add a lock on read_events and Catalog.copy wouldn't it drastically reduce the benefit you get from multi-threading for solving the small corner case of duplicate resource_ids? If we do create a decorator for thread locking, where should such a decorator live, in obspy.util.decorator? 

I dont understand how the race condition is created with the ResourceIdentifier methods you mentioned, although there probably is one I am just not seeing, as this is getting rather complicated. Thoughts?